### ISSUE TITLE
Clipboard error : Target STRING not available

### ISSUE THREAD

**Issue by rtucek** [svermeulen/vim-yoink#16]
Hi,

I've just started using the plugin. When starting neovim, I get the error `Clipboard error : Target STRING not available`. This only affects neovim, but not vim.

Googling around, I found https://github.com/neovim/neovim/issues/2642#issuecomment-218232937, which relates to logic of monitoring the system's clipboard via yankrinkg.

And indeed, setting `let g:yoinkSyncSystemClipboardOnFocus = 0` prevents this error.
Is there a way of having this fixed, while retaining the monitoring of the system's clipboard?

**Comment by rtucek** [comment]
And yes, I'm using `xclip`.

**Comment by svermeulen** [comment]
What version of neovim?

**Comment by rtucek** [comment]
Hi @svermeulen, thank you for looking into this!

I use Neovim v0.4.3

However, I think I've made some mistake. I claimed `let g:yoinkSyncSystemClipboardOnFocus = 0` would prevent showing up the error. However, I noticed that even having the setting disabled, the error shows up occasionally.

**Comment by svermeulen** [comment]
I can't seem to reproduce this.  I am on osx though so maybe it doesn't appear there.  I am guessing that it occurs during the `yoink#onVimEnter` method though based on your description.  If it's annoying enough, you could uncomment the `yoink#addCurrentDefaultRegToHistory` line in `yoink#onVimEnter` since I don't expect that is totally necessary

**Comment by rtucek** [comment]
I've done more googling on the error and I think it's exclusively related to xclip (which is for interacting with the clipboard on Linux). So I doubt you'll encounter this on MacOS either. The issues I found don't relate to neovim/vim, but arise with similar interaction with the clipboard.

If you don't mind, I'll keep this issue open for more try'n'error feedback. I'll try out your suggested fix this weekend.

**Comment by yigitsever** [comment]
I had this error for ages, https://github.com/neovim/neovim/issues/2642#issuecomment-170271675 solved it for me so I believe xclip is the culprit.

BUT, xclip might be a dependency and (at least for nvim) it loads before xsel `:help clipboard-tool`.

So; setting the clipboard provider to xsel explicitly `let g:clipboard = "xsel"` solves the issue if for some reason you need to have xclip.

**Comment by yigitsever** [comment]
Seems like xclip is indeed the issue, removing it and using xsel fixed it for me but I have a xclip dependency. nvim tries to use xclip before xsel `:h clipboard-tool`. You can manually override the clipboard tool using;

```
let g:clipboard = {
      \   'name': 'xsel_override',
      \   'copy': {
      \      '+': 'xsel --input --clipboard',
      \      '*': 'xsel --input --primary',
      \    },
      \   'paste': {
      \      '+': 'xsel --output --clipboard',
      \      '*': 'xsel --output --primary',
      \   },
      \   'cache_enabled': 1,
      \ }
```
if you need to have both xsel and xclip AND you're using nvim. This snippet works with `set clipboard+=unnamedplus` as advised in `:h clipboard`.

**Comment by rtucek** [comment]
I haven't tried @yigitsever's [solution](https://github.com/svermeulen/vim-yoink/issues/16#issuecomment-632234373), even several people report it fixed the issue for them. The main reasons are that I did not want to have xsel installed along xclip as well as the error wasn't that much an issue for me given it only happened sporadic.

However, after upgrading neovim I started encountering the error as reported in #28.

I think this is related to this issue... previously it just "used" to issue the small error `Clipboard error : Target STRING not available` but the more recent neovim version seems to have more troubles with it. I'm closing this issue though as we should continue with the thread in #28.
